soc/raspberrypi/rpi_pico/soc.yml

```diff
@@ -1,6 +1,9 @@
 family:
 - name: rpi_pico
   series:
-  - name: rp2xxx
+  - name: rp2040
     socs:
     - name: rp2040
+  - name: rp2350
+    socs:
+    - name: rp2350a
```

Comment:
add CPU cluster (since only ARM is added here, the riscv one can be left off)

Reply:
One question.
RP2350 has two independent Cortex-M33.
In this case, should they be divided into `m33_0` and `m33_1`?

Reply:
They can be yes

Reply:
@nordicjm 
Thank you for your answer.

@ajf58 
Let's proceed with this approach. The other addressed issue should also be resolved with this.

Reply:
So the needed addition would be something like that?
```
            cpuclusters:
              - name: m33_0 
              - name: m33_1
```

Reply:
> So the needed addition would be something like that?

Yes, that's what I think.

Reply:
That would be fine

Reply:
Ah, I got it. My suggestion was wrong.
I think no need to define RISC-V part.
So it finally should be `m33` as you suggest. It is acceptable for me.

Reply:
I think the definition of ‘cpuclusters‘ should be added,
but at the moment, it has not been implemented throughout the system,
so I think it can be excluded from this PR for now.

@ajf58 
Please consider addressing a separate PR.

Reply:
This is not resolved, as said it needs to have the CPU cluster. This is a multi-core device, someone *might* (in the future) build for one of 4 cores, and they need to be able to do that, so the **board** files and **soc** files **must** have the CPU cluster in and be named for the single one that is supported, including Kconfigs, if only one is currently supported then you only add one. The board.yml file lists the SoC, the Kconfig in the board file selects the exact core Kconfig which needs to match up with the values in soc.yml and soc Kconfig files. By breaking things I mean if you add it wrong, as it currently is, you cause mass breakage **for everyone** using this SoC when support is added for another core because suddenly their board names are wrong, their Kconfig selections are wrong, their board filenames are wrong

Reply:
> someone might (in the future) build for one of 4 cores, and they need to be able to do that, so the board files and soc files must have the CPU cluster in and be named for the single one that is supported, including Kconfigs, if only one is currently supported then you only add one

I disagree with the approach that's been strongly pushed for here.

The original objective was to get a MVP onto `main`. This is speculating about what someone wants/needs, and also what the rest of the Zephyr project is willing to support. The RP2040 on the Pico 1 has two fixed Cortex-M0+ cores, and  there's no support ~years on for SMP or even just running a function on the second core. So, does that (the RP2040 and Pico 1) need fixing? Not in my mind, it's fine, just as the RP2350A was. Was this a better use of time than adding support for secure vs non-secure? Who knows. What would've happened if another reviewer NACKed this review because I haven't implemented that? Again, who knows.

> By breaking things I mean if you add it wrong, as it currently is, you cause mass breakage for everyone using this SoC when support is added for another core because suddenly their board names are wrong, their Kconfig selections are wrong, their board filenames are wrong

"mass breakage" feels a lot like hyperbole, and detracts from the discussion. I feel it's important to be mindful that It wouldn't break all by itself. At the most basic, the user chooses to pull changes from `main`; they're not just floating onto their computer in the ether. As developers we can make it our responsibility to maintain good quality documentation (including the migration guide and deprecation warnings, for example), and likewise it's our responsibility to read those documents.

Reply:
I would also like the configuration here to be the same as Pico1.
Adding SMP is likely to be a major change, so I am concerned that it will affect the interface, etc. I think it will affect users, but I think the `migration-guide` is there to document that and handle it correctly.

Reply:
> The RP2040 on the Pico 1 has two fixed Cortex-M0+ cores, and there's no support ~years on for SMP or even just running a function on the second core. So, does that (the RP2040 and Pico 1) need fixing?

Multi-core support is not an ARM thing with cortex m0, there are 2 vendors in the world that support it, one is raspberry pi and the other is sony, both have completely custom silicon for it because as said, it is not part of the cortex m0 (armv6) architecture, there is support in zephyr because it would be very complex to support and is very bespoke. It is also SMP, *not* AMP, so you don't load 2 different images, you load a single image.
For the rp2350 it is 2 independent cores, and for those cores it can be m33 or riscv. Now I've not looked into if it supports SMP or AMP but based on it supporting 2 different architectures for each core I'm going to assume it's AMP, so that needs to be represented properly here. If this whole chip is SMP only then this is OK, if this supports AMP then the cpucluster needs to be changed to e.g. `m33_0`.

> "mass breakage" feels a lot like hyperbole, and detracts from the discussion. I feel it's important to be mindful that It wouldn't break all by itself. At the most basic, the user chooses to pull changes from main; they're not just floating onto their computer in the ether. As developers we can make it our responsibility to maintain good quality documentation (including the migration guide and deprecation warnings, for example), and likewise it's our responsibility to read those documents.

It's to highlight that it would be a bad implementation, a good implementation plans for the future, it's not like you don't know there are multiple cores in this chip, it's not like the vendor has never mentioned it then in a year just suddenly announces that that hardware supported it all along but they're only announcing it now.

Reply:
Will accept m33 for now, have talked about issues which need fixing for a future improvement to the hardware model